demos/coreMQTT/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1959 @@
+            xNetworkConnectionCreated = pdTRUE;
+            /* Form an MQTT connection with a persistent session. */
+            xMQTTStatus = prvMQTTConnect( &globalMqttContext, &xNetworkContext, false );
+            ret = ( xMQTTStatus == MQTTSuccess ) ? EXIT_SUCCESS : EXIT_FAILURE;
+        }
+
+        if( ret == EXIT_SUCCESS )
+        {
+            configASSERT( globalMqttContext.connectStatus == MQTTConnected );
+
+            /* Give subscriber task higher priority so the subscribe will be processed before the first publish.
+             * This must be less than or equal to the priority of the main task. */
+            xResult = xTaskCreate( prvSubscribeTask, "Subscriber", mqttexampleTASK_STACK_SIZE, NULL, tskIDLE_PRIORITY + 1, &xSubscribeTask );
+            ret = ( xResult == pdPASS ) ? EXIT_SUCCESS : EXIT_FAILURE;
+        }
+
+        if( ret == EXIT_SUCCESS )
+        {
+            xResult = xTaskCreate( prvPublishTask, "Publisher", mqttexampleTASK_STACK_SIZE, NULL, tskIDLE_PRIORITY, &xPublisherTask );
+            ret = ( xResult == pdPASS ) ? EXIT_SUCCESS : EXIT_FAILURE;
+        }
+
+        if( ret == EXIT_SUCCESS )
+        {
+            LogInfo( ( "Running command loop" ) );
+            ret = prvCommandLoop();
+        }
+
+        if( ret == EXIT_SUCCESS )
+        {
+            /* Wait for tasks to exit before cleaning up. */
+            while( ( ulNotification & mqttexampleSUBSCRIBE_TASK_COMPLETE_BIT ) != mqttexampleSUBSCRIBE_TASK_COMPLETE_BIT )
+            {
+                LogInfo( ( "Waiting for subscribe task to exit." ) );
+                xTaskNotifyWait( 0, mqttexampleSUBSCRIBE_TASK_COMPLETE_BIT, &ulNotification, mqttexampleDEMO_TICKS_TO_WAIT );
+            }
+
+            /* Wait for publishing task to exit before cleaning up. */
+            while( ( ulNotification & mqttexamplePUBLISHER_TASK_COMPLETE_BIT ) != mqttexamplePUBLISHER_TASK_COMPLETE_BIT )
+            {
+                LogInfo( ( "Waiting for publish task to exit." ) );
+                xTaskNotifyWait( 0, mqttexamplePUBLISHER_TASK_COMPLETE_BIT, &ulNotification, mqttexampleDEMO_TICKS_TO_WAIT );
+            }
+
+            /* Reset queues. */
+            xQueueReset( xCommandQueue );
+            xQueueReset( xPublisherResponseQueue );
+            xQueueReset( xSubscriberResponseQueue );
+        }
+
+        /* Close network connection even if failure occurred elsewhere. */
+        if( xNetworkConnectionCreated == pdTRUE )
+        {
+            LogInfo( ( "Disconnecting TCP connection." ) );
+            xNetworkStatus = prvDisconnectNetwork( &xNetworkContext );
+            ret = ( xNetworkStatus == pdPASS ) ? EXIT_SUCCESS : EXIT_FAILURE;
+            xNetworkConnectionCreated = pdFALSE;
+        }
+
+        if( ret == EXIT_SUCCESS )
+        {
+            LogInfo( ( "RunCoreMqttConnectionSharingDemo() completed an iteration successfully. Total free heap is %u.\r\n",
+                       xPortGetFreeHeapSize() ) );
```

Comment:
Let's delete this xPortGetFreeHeapSize(). 
```suggestion
            LogInfo( ( "RunCoreMqttConnectionSharingDemo() completed an iteration successfully." ) ) );
```

Reply:
Changed